tools/src/main/java/org/apache/kafka/tools/reassign/ReassignPartitionsCommand.java

```diff
@@ -0,0 +1,1499 @@
+            adminClient = Admin.create(props);
+            handleAction(adminClient, opts);
+            failed = false;
+        } catch (TerseException e) {
+            System.out.println(e.getMessage());
+        } catch (Throwable e) {
+            System.out.println("Error: " + e.getMessage());
+            System.out.println(Utils.stackTrace(e));
+        } finally {
+            // It's good to do this after printing any error stack trace.
+            if (adminClient != null) {
+                adminClient.close();
+            }
+        }
+        // If the command failed, exit with a non-zero exit code.
+        if (failed) {
+            Exit.exit(1);
+        }
+    }
+
+    private static void handleAction(Admin adminClient, ReassignPartitionsCommandOptions opts) throws IOException, ExecutionException, InterruptedException, TerseException {
+        if (opts.options.has(opts.verifyOpt)) {
+            verifyAssignment(adminClient,
+                Utils.readFileAsString(opts.options.valueOf(opts.reassignmentJsonFileOpt)),
+                opts.options.has(opts.preserveThrottlesOpt));
+        } else if (opts.options.has(opts.generateOpt)) {
+            generateAssignment(adminClient,
+                Utils.readFileAsString(opts.options.valueOf(opts.topicsToMoveJsonFileOpt)),
+                opts.options.valueOf(opts.brokerListOpt),
+                !opts.options.has(opts.disableRackAware));
+        } else if (opts.options.has(opts.executeOpt)) {
+            executeAssignment(adminClient,
+                opts.options.has(opts.additionalOpt),
+                Utils.readFileAsString(opts.options.valueOf(opts.reassignmentJsonFileOpt)),
+                opts.options.valueOf(opts.interBrokerThrottleOpt),
+                opts.options.valueOf(opts.replicaAlterLogDirsThrottleOpt),
+                opts.options.valueOf(opts.timeoutOpt),
+                Time.SYSTEM);
+        } else if (opts.options.has(opts.cancelOpt)) {
+            cancelAssignment(adminClient,
+                Utils.readFileAsString(opts.options.valueOf(opts.reassignmentJsonFileOpt)),
+                opts.options.has(opts.preserveThrottlesOpt),
+                opts.options.valueOf(opts.timeoutOpt),
+                Time.SYSTEM);
+        } else if (opts.options.has(opts.listOpt)) {
+            listReassignments(adminClient);
+        } else {
+            throw new RuntimeException("Unsupported action.");
+        }
+    }
+
+    /**
+     * The entry point for the --verify command.
+     *
+     * @param adminClient           The AdminClient to use.
+     * @param jsonString            The JSON string to use for the topics and partitions to verify.
+     * @param preserveThrottles     True if we should avoid changing topic or broker throttles.
+     *
+     * @return                      A result that is useful for testing.
+     */
+    static VerifyAssignmentResult verifyAssignment(Admin adminClient,
```

Comment:
Hi @nizhikov, I found many methods are static methods (which means default permission), our application uses the function of partition assignment(not script), and we are using the partition assignment in our app by scala API. but now we can't access these static methods. 


```
verifyAssignment(org.apache.kafka.clients.admin.Admin, java.lang.String, java.lang.Boolean)' is not public in 'org.apache.kafka.tools.reassign.ReassignPartitionsCommand'. Cannot be accessed from outside package
```

Reply:
Yes. It's an expected side effect.

Methods of this class are not public API and can be changed without any restrictions.
I advice you to modify your code.

Reply:
ok, Thank for your reply